core/src/main/java/org/elasticsearch/index/IndexService.java

```diff
@@ -140,6 +145,7 @@ public IndexService(IndexSettings indexSettings, NodeEnvironment nodeEnv,
         this.listeners = new IndexingOperationListener[1+listenersIn.length];
         this.listeners[0] = slowLog;
         System.arraycopy(listenersIn, 0, this.listeners, 1, listenersIn.length);
+        this.refreshTask = new RefreshTasks(this, nodeServicesProvider.getThreadPool());
```

Comment:
Can we call `getThreadPool()` in all these places that are now calling `nodeServicesProvider.getThreadPool()`?  Or maybe have a `this.threadPool` instead?